blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,887 @@
+      23: 0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+
+3. Run `extend_all.sh` script from `landing-zone` package.
+
+This script simulates what should be extended into PCR17 by SKINIT, LZ and
+kernel during platform booting. It extends both SHA256 and SHA1 values. However,
+expected result is valid only for SHA256 if used with TPM2.0 device.
+
+To properly execute script, first find correct directory to `bzImage` and
+`initrd`. Best way to find exact directories is to see `"NixOS - Secure Launch"`
+entry in `/boot/grub/grub.cfg`:
+
+    ```
+    $ cat /boot/grub/grub.cfg
+    (...)
+    menuentry "NixOS - Secure Launch" {
+      search --set=drive1 --fs-uuid fcc62677-b961-4ccf-bd66-376db104240f
+      search --set=drive2 --fs-uuid fcc62677-b961-4ccf-bd66-376db104240f
+      slaunch skinit
+      slaunch_module ($drive2)/boot/lz_header
+      linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/1mgqiy35hksf0r66gfffrl76s2img9z2-nix4
+        initrd ($drive2)/nix/store/gyqhrgvapfhfqq8x1km3z9ipv7phcadq-initrd-linux-5.1.0/initrd      
+    }
+    (...)
+    ```
+
+`/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage` is directory
+to Linux kernel.
+`/nix/store/gyqhrgvapfhfqq8x1km3z9ipv7phcadq-initrd-linux-5.1.0/initrd` is
+directory to initrd.
+
+4. Go to `/nix/store/` and run below command:
+
+    ```
+    $ cd /nix/store
+    $ ls | grep landing-zone
+    5q92f6l4s1jfbw5ygfr1sd4hlczjj6l2-landing-zone-0.3.0.drv
+    6v15ikqsyqk5fs0jg1n6755dp1nr6cyc-landing-zone-debug-0.3.0.drv
+    dnpqvb64jjr3x2kxx92wvdkvmah72h6m-landing-zone-debug-0.3.0
+    zpcf7yf1fjf9slz2sr2f6s3wl3ch1har-landing-zone-0.3.0
+    ```
+
+>Hash before `-landing-zone-1.0` is dependent on built version and might be
+different in yours. Choose non-debug version from above results.
+
+5. Go to `/nix/store/zpcf7yf1fjf9slz2sr2f6s3wl3ch1har-landing-zone-0.3.0`
+directory.
+
+    ```
+    $ cd /nix/store/zpcf7yf1fjf9slz2sr2f6s3wl3ch1har-landing-zone-0.3.0
+    ```
+
+6. Execute `./extend_all.sh` script.
+
+Usage is `./extend_all.sh <directory-to-bzImage> <directory-to-initrd>`
+
+It must be executed inside directory containing `lz_header.bin`. You should
+already be in this directory after previous step. Directories to `bzImage` and
+`initrd` we found in step 3.
+
+    ```
+    ./extend_all.sh /nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage /nix/store/gyqhrgvapfhfqq8x1km3z9ipv7phcadq-initrd-linux-5.1.0/initrd
+    d91e7f685bcae20f84308eafe46f02eea8fcc90c  SHA1
+    7392be6cd449323115d11bbc97af4cb2adad25b9cf52d0861f87934feea7b03e  SHA256
+    ```
+
+Compare SHA256 value with PCR17 content checked previously with `tpm2_pcrread`
+output. If DRTM is enabled and executes properly, they should be the same. It
+proves that LZ code utilizes SHA256 algorithm during measurements.
+
+##### Check if LZ debug option can be enabled
+
+1. Boot NixOS and go to `/nix/store/` directory.
```

Comment:
Are points 1-5 really necessary? They are identical as previous, you can just point to that section

Reply:
I rather would leave it this way. I know what you mean, but sometimes it is easier to just write those few steps then points to another line in document and go back here.

Reply:
Ok, fair point